plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubActionsDispatch.ts

```diff
@@ -13,27 +13,15 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { InputError } from '@backstage/errors';
-import {
-  GithubCredentialsProvider,
-  ScmIntegrationRegistry,
-} from '@backstage/integration';
-import { Octokit } from '@octokit/rest';
-import { parseRepoUrl } from '../publish/util';
+import { ScmIntegrationRegistry } from '@backstage/integration';
 import { createTemplateAction } from '../../createTemplateAction';
+import { getOctokit } from './helpers';
 
 export function createGithubActionsDispatchAction(options: {
   integrations: ScmIntegrationRegistry;
 }) {
   const { integrations } = options;
 
-  const credentialsProviders = new Map(
```

Comment:
Awesome to see cleanups 💯 

One issue with this is that we no longer get the credentials caching as we're recreating the GithubCredentialsProvider inside the method on every call. Could we do something better? Perhaps a class or passing it in?

Reply:
@jhaals I have introduced OctokitProvider class in helpers to allow for caching credentials.

Reply:
Nice, that looks a lot better to me 👍 

Perhaps worth renaming helpert.ts to OctokitProvider.ts now that it's basically one class instead of a file for utility methods. Some docstrings on the class and method inside would be great too.

Reply:
Done :)